### PR TITLE
Create `BaseMultiTableSynthesizer` and `HMASynthesizer`

### LINKED ISSUE #1070: Create BaseMultiTableSynthesizer and HMASynthesizer classes
### Problem Description

_As a user, it would be useful to be able to synthesize data at the multi-table level. Additionally, it would be helpful to manage all of the transformations and modeling for each table in one place._

### Acceptance Criteria

- There should be a new module in `SDV` called `multi_table`
- There should be a `BaseMultiTableSynthesizer` class that lives in the `multi-table` module
- There should be a class called `HMASynthesizer` that inherits from the `BaseMultiTableSynthesizer`
- The `__init__` should have the following parameters:
    - `metadata`: A `MultiTableMetadata` object
- The base class should have four additional methods:
    1. `get_table_parameters(table_name)`: Get the parameters for a given table name
    2. `update_table_parameters(table_name, table_parameters)`: Update the parameters for a given table
    3. `get_metadata()`: Returns the `MultiTableMetadata` object
    4. `get_parameters(table_name)`: Returns the parameters that were used to instantiate the synthesizer for the given table.

### Expected behavior
```python
>>> synthesizer = HMASynthesizer(metadata)

>>> synthesizer.get_table_parameters(table_name='users')
{
  'enforce_min_max_values': True,
  'enforce_rounding': True,
  'numerical_distributions': {},
  'default_distribution': 'beta',
}

>>> synthesizer.update_table_parameters(
  table_name='users',
  table_parameters={
    'enforce_min_max_values': False,
    'numerical_distributions': {
      'age': 'uniform',
      'height': 'gaussian'
    }
  }
)

>>> metadata = synthesizer.get_metadata()
>>> print(metadata)
{
  "tables": {
    "accounts": { <Table Spec Above> },
    "branches": { <Table Spec Above> }
  }, 
  "relationships": [
    ...
  ],
}

>>> synthesizer.get_parameters(table_name='users')
{
  'enforce_min_max_values': True,
  'enforce_rounding': False,
  'numerical_column_distributions': {
    'age': 'beta',
    'height': 'gaussian'
  }
}
```

### Additional context

- Cross check with the current [HMA](https://github.com/sdv-dev/SDV/blob/master/sdv/relational/hma.py) class to see any additional attributes that may be required. We will need a dictionary that maps table names to the synthesizer used for that table. `HMA` only supports the `GaussianCopulaSynthesizer`

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #1070

### CODE REVIEW

File: sdv/data_processing/data_processor.py [hunk 1]
```diff
@@ -375,7 +375,7 @@ def prepare_for_fitting(self, data):
                 'Setting the configuration for the ``HyperTransformer`` '
                 f'for table {self.table_name}'
             ))
-            config = self._create_config(data, columns_created_by_constraints)
+            config = self._create_config(constrained, columns_created_by_constraints)
```
Comment (amontanez24) [inline sdv/data_processing/data_processor.py:378]: this should go away after rebasing right?

File: sdv/multi_table/hma.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    Args:
+        metadata (dict, str or Metadata):
+            Metadata dict, path to the metadata JSON file or Metadata instance itself.
+    """
+
+    DEFAULT_SYNTHESIZER_KWARGS = {
+        'default_distribution': 'norm',
+    }
+    _synthesizer = GaussianCopulaSynthesizer
```
Comment (amontanez24) [inline sdv/multi_table/hma.py:18]: is this needed since it is the same as the base?

File: sdv/multi_table/base.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        """Update the table's synthesizer instantiation parameters.
+
+        Args:
+            table_name (str):
+                Table name for which the parameters should be retrieved.
+            table_parameters (dict):
+                A dictionary with the parameters as keys and the values to be used to instantiate
+                the table's synthesizer.
+        """
+        self._table_parameters[table_name].update(deepcopy(table_parameters))
```
Comment (amontanez24) [inline sdv/multi_table/base.py:76]: can we move this line below the one under it in case the initialization crashes?

**Review by amontanez24 — APPROVED** [review]
A few minor comments, but for the most part this LGTM!

File: tests/unit/multi_table/test_base.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    def test___init__(self):
+        """Test that when creating a new instance this sets the defaults.
+
+        Test that the metadata object is being stored and also being validated. Afterwards, this
+        calls the ``self._initialize_models`` which creates the initial instances of those.
+        """
+        # Setup
+        metadata = get_multi_table_metadata()
+        metadata.validate = Mock()
+        # Run
```
Comment (fealho) [inline tests/unit/multi_table/test_base.py:51]: Missing \n

**Review by fealho — APPROVED** [review]
Looks good  👍

**Pull request merged**